repo/manifest/manifest_manager.go

```diff
@@ -161,6 +150,11 @@ func findEntriesMatchingLabels(m map[ID]*manifestEntry, labels map[string]string
 
 // Find returns the list of EntryMetadata for manifest entries matching all provided labels.
 func (m *Manager) Find(ctx context.Context, labels map[string]string) ([]*EntryMetadata, error) {
+	committedMatches, err := m.committed.findCommittedEntries(ctx, labels)
```

Comment:
Same as above, this is implicitly assuming the entries in `commitedMatches` are not modified.